utils.go

```diff
@@ -257,3 +258,39 @@ func FisherYatesShuffle(inputs []interface{}, nhash ...uint64) {
 		inputs[i], inputs[j] = inputs[j], inputs[i]
 	}
 }
+
+func delve(obj map[string]interface{}, key string) (interface{}, bool) {
+	props := strings.Split(key, ".")
+	var result interface{}
+	result, ok := obj[props[0]]
+	if ok {
+		if len(props) > 1 {
+			i := 0
+			newProps := append(props[:i], props[i+1:]...)
+			result, _ = delve(result.(map[string]interface{}), strings.Join(newProps, "."))
+		}
+	}
+	return result, result != nil
+}
+
+func delveCreate(obj map[string]interface{}, key string, val interface{}) bool {
+	props := strings.Split(key, ".")
+
+	if len(props) == 0 {
+		return false
+	} else if len(props) == 1 {
+		obj[key] = val
+		return true
```

Comment:
line 279-283 can be condensed into:
if len(props) < 2 {
  return (len(props) == 1)
}